### PR TITLE
MEPTS-1362 | RESUMO MENSAL DE DAH

### PR DESCRIPTION
**Pull request by helderjosue**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/advancedhivillness/ResumoMensalDAHCohortQueries.java [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+    map.put("1707", hivMetadata.getAbandonedConcept().getConceptId());
+    map.put("1708", hivMetadata.getExitFromArvTreatmentConcept().getConceptId());
+    map.put("165386", hivMetadata.getExitDateFromArvTreatmentConcept().getConceptId());
+
+    String query =
+        "SELECT p.patient_id "
+            + "FROM "
+            + "    patient p INNER JOIN encounter e ON p.patient_id = e.patient_id "
+            + "              INNER JOIN obs o on e.encounter_id = o.encounter_id "
+            + "              INNER JOIN obs o2 on e.encounter_id = o2.encounter_id "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/advancedhivillness/ResumoMensalDAHCohortQueries.java]: @helderjosue You don't need this o2, it is going to slow your query

**Review by psmatsinhe — CHANGES_REQUESTED** [review]
@helderjosue Check the commets

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/advancedhivillness/ResumoMensalDAHCohortQueries.java (hunk 1 above)
Comment (helderjosue) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/advancedhivillness/ResumoMensalDAHCohortQueries.java]: Noted and Corrected!

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/resumo/ResumoMensalDAHQueries.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        + valueNumeric
+        + " AND ( "
+        + "  ( o.obs_datetime >= :startDate "
+        + "  AND o.obs_datetime <= :endDate)"
+        + "OR "
+        + " ( e.encounter_datetime >= :startDate "
+        + "  AND e.encounter_datetime <= :endDate) "
+        + " ) "
+        + "  AND e.location_id = :location"
+        + "  AND TIMESTAMPDIFF(YEAR, ps.birthdate, :endDate) >=5 "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/resumo/ResumoMensalDAHQueries.java]: @helderjosue acho melhor usar a idade em uma composition e nao aqui

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/resumo/ResumoMensalDAHQueries.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        + " AND ( "
+        + "  ( o.obs_datetime >= :startDate "
+        + "  AND o.obs_datetime <= :endDate)"
+        + "OR "
+        + " ( e.encounter_datetime >= :startDate "
+        + "  AND e.encounter_datetime <= :endDate) "
+        + " ) "
+        + "  AND e.location_id = :location"
+        + " AND TIMESTAMPDIFF(YEAR, ps.birthdate, :endDate) >=1 "
+        + " AND TIMESTAMPDIFF(YEAR, ps.birthdate, :endDate) <5 "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/resumo/ResumoMensalDAHQueries.java]: Idem

**Review by psmatsinhe — CHANGES_REQUESTED** [review]
@helderjosue please have a look

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/resumo/ResumoMensalDAHQueries.java (hunk 2 above)
Comment (helderjosue) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/resumo/ResumoMensalDAHQueries.java]: Noted and Corrected!

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/resumo/ResumoMensalDAHQueries.java (hunk 3 above)
Comment (helderjosue) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/resumo/ResumoMensalDAHQueries.java]: Noted and Corrected!

**Review by psmatsinhe — APPROVED** [review]
Looks good to me

**Pull request merged**